### DISCUSSION TITLE
connect budibase with database

### DISCUSSION THREAD

**Discussion by LTCTHITA** [Budibase/budibase#13037] [Help]
Hey,
 
I want to connect to SQL server with Budibase.
 But budibase shows error: 'Error creating datasource: Unable to connect: Failed to connect to 192.168.69.154:1433 in 15000ms'
 I've tried changing internet networks and it's still the same error.
Check Firewall and connect normally.
But I can connect between Database and PowerBi.
Can you tell me how to fix this problem?

![Screenshot 2024-02-15 081332](https://github.com/Budibase/budibase/assets/93240913/2e76b3e6-d53a-4489-8230-595270104fc4)

![Screenshot 2024-02-15 084119](https://github.com/Budibase/budibase/assets/93240913/4325a00b-e5a6-434b-9031-3312303d708f)

![Screenshot 2024-02-15 085026](https://github.com/Budibase/budibase/assets/93240913/9e2027d5-ff98-4c36-a870-106874d712b8)

![Screenshot 2024-02-15 085008](https://github.com/Budibase/budibase/assets/93240913/43b28e4f-cea2-4c70-ae14-d573be4702b9)
 
Thank you,

**Comment by melohagan** [comment]
Hey @LTCTHITA 

Are you self-hosting? You would need to be in order to connect to 192 addresses.

As Budibase is run via Docker, you may need to to configure your docker network settings, port forwarding, etc.
